src/components/icon/icon-registry.ts

```diff
@@ -0,0 +1,370 @@
+   * from it.
+   */
+  private _loadSvgIconFromConfig(config: SvgIconConfig): Observable<SVGElement> {
+    return this._fetchUrl(config.url)
+        .map(svgText => this._createSvgElementForSingleIcon(svgText, config));
+  }
+
+  /**
+   * Loads the content of the icon set URL specified in the SvgIconConfig and creates an SVG element
+   * from it.
+   */
+  private _loadSvgIconSetFromConfig(config: SvgIconConfig): Observable<SVGElement> {
+    return this._fetchUrl(config.url)
+        .map((svgText) => this._svgElementFromString(svgText));
+  }
+
+  /**
+   * Creates a DOM element from the given SVG string, and adds default attributes.
+   */
+  private _createSvgElementForSingleIcon(responseText: string, config: SvgIconConfig): SVGElement {
+    const svg = this._svgElementFromString(responseText);
+    this._setSvgAttributes(svg, config);
+    return svg;
+  }
+
+  /**
+   * Searches the cached element of the given SvgIconConfig for a nested icon element whose "id"
+   * tag matches the specified name. If found, copies the nested element to a new SVG element and
+   * returns it. Returns null if no matching element is found.
+   */
+  private _extractSvgIconFromSet(
+      iconSet: SVGElement, iconName: string, config: SvgIconConfig): SVGElement {
+    const iconNode = iconSet.querySelector('#' + iconName);
+    if (!iconNode) {
+      return null;
+    }
+    // If the icon node is itself an <svg> node, clone and return it directly. If not, set it as
+    // the content of a new <svg> node.
+    if (iconNode.tagName.toLowerCase() == 'svg') {
+      return this._setSvgAttributes(<SVGElement>iconNode.cloneNode(true), config);
+    }
+    // createElement('SVG') doesn't work as expected; the DOM ends up with
+    // the correct nodes, but the SVG content doesn't render. Instead we
+    // have to create an empty SVG node using innerHTML and append its content.
+    // http://stackoverflow.com/questions/23003278/svg-innerhtml-in-firefox-can-not-display
+    const svg = this._svgElementFromString('<svg></svg>');
+    // Clone the node so we don't remove it from the parent icon set element.
+    svg.appendChild(iconNode.cloneNode(true));
+    return this._setSvgAttributes(svg, config);
+  }
+
+  /**
+   * Creates a DOM element from the given SVG string.
+   */
+  private _svgElementFromString(str: string): SVGElement {
+    // TODO: Is there a better way than innerHTML? Renderer doesn't appear to have a method for
+    // creating an element from an HTML string.
+    const div = document.createElement('DIV');
+    div.innerHTML = str;
+    const svg = <SVGElement>div.querySelector('svg');
+    if (!svg) {
+      throw new MdIconSvgTagNotFoundException();
+    }
+    return svg;
+  }
+
+  /**
+   * Sets the default attributes for an SVG element to be used as an icon.
+   */
+  private _setSvgAttributes(svg: SVGElement, config: SvgIconConfig): SVGElement {
+    if (!svg.getAttribute('xmlns')) {
```

Comment:
This can go through the renderer, e.g. `this._renderer.getElementAttribute` and  `this._renderer.setElementAttribute`